### PR TITLE
Obstecle course first

### PR DESCRIPTION
**Pull request by OLeg25685**
1- go forward from start point (-60, 60, 270 degrees)
2- turn along the arc 
3- parking on coordinates (5, 10, 0 degrees))
4- there is full path
image

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/OlehPath.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        Trajectory trajectorySecondPart = drive.trajectoryBuilder(trajectoryFirstPart.end())
+                .splineTo(new Vector2d(7.5, 12), Math.toRadians(0))
+                .build();
+
+
+
+
+
+
```
Comment (CubeOfCheese) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/OlehPath.java]: get rid of the extra lines

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/OlehPath.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import com.acmerobotics.roadrunner.geometry.Vector2d;
+import com.acmerobotics.roadrunner.trajectory.Trajectory;
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+
+import org.firstinspires.ftc.teamcode.drive.SampleMecanumDrive;
+
+@Autonomous (group = "drive")
+
+public class OlehPath extends LinearOpMode{
```
Comment (CubeOfCheese) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/OlehPath.java]: Don't name the class or file after yourself. It should be named something like ObstacleCoursePath

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/OlehPath.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+    public void runOpMode() {
+        SampleMecanumDrive drive = new SampleMecanumDrive(hardwareMap);
+
+        drive.setPoseEstimate(new Pose2d(-60, 60,Math.toRadians(270)));
```
Comment (CubeOfCheese) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/OlehPath.java]: space after comma please
```suggestion
        drive.setPoseEstimate(new Pose2d(-60, 60, Math.toRadians(270)));
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/OlehPath.java (hunk 2 above)
Comment (CubeOfCheese) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/OlehPath.java]: space before curly brace please
```suggestion
public class OlehPath extends LinearOpMode {
```

**Review by CubeOfCheese — CHANGES_REQUESTED** [review]

**Review by isabellammjfk — APPROVED** [review]
looks good and it works!

**Review by CubeOfCheese — APPROVED** [review]

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java [hunk 4]
```diff
@@ -8,16 +8,25 @@
...
 
+        drive.setPoseEstimate(new Pose2d(-60, 60, Math.toRadians(270)));
 
-        SampleMecanumDrive drive = new SampleMecanumDrive(hardwareMap);
+        Trajectory trajectoryFirstPart = drive.trajectoryBuilder(new Pose2d(-60, 60, Math.toRadians(270)))
+                .splineTo(new Vector2d(-57.3, 29), Math.toRadians(270))
+                .build();
+
+        Trajectory trajectorySecondPart = drive.trajectoryBuilder(trajectoryFirstPart.end())
+                .splineTo(new Vector2d(7.5, 12), Math.toRadians(0))
```
Comment (AkithMadurapperuma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java]: Kevina's code (2nd Chunk) starts at (5, 10) while yours ends at (7.5, 12). This should be fixed.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java [hunk 5]
```diff
@@ -9,12 +9,24 @@
 import org.firstinspires.ftc.teamcode.trajectorysequence.TrajectorySequence;
 
 @Autonomous (group = "drive")
+
 public class ObstacleCoursePath extends LinearOpMode {
+
```
Comment (CubeOfCheese) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java]: Get rid of these extra lines

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java [hunk 6]
```diff
@@ -52,6 +60,9 @@ public void runOpMode() {
 
         if(isStopRequested()) return;
 
+        // first chunk
+        drive.followTrajectory(trajectoryFirstPart);
```
Comment (isabellammjfk) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java]: maybe change the trajectory names of your path to match where you're going on the field

**Review by isabellammjfk — COMMENTED** [review]
good job!!

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java [hunk 7]
```diff
@@ -14,16 +14,24 @@ public class ObstacleCoursePath extends LinearOpMode {
...
+        drive.setPoseEstimate(new Pose2d(-60, 60, Math.toRadians(270)));
+        Trajectory trajectoryFirstPart = drive.trajectoryBuilder(new Pose2d(-60, 60, Math.toRadians(270)))
+                .splineTo(new Vector2d(-57.3, 29), Math.toRadians(270))
+                .build();
+        Trajectory trajectorySecondPart = drive.trajectoryBuilder(trajectoryFirstPart.end())
+                .splineTo(new Vector2d(5, 10), Math.toRadians(0))
+                .build();
 
         // second part
+        drive.setPoseEstimate(new Pose2d(5, 10, Math.toRadians(0)));
```
Comment (CubeOfCheese) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java]: We should only have one setPoseEstimate

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java [hunk 8]
```diff
@@ -14,8 +14,14 @@ public class ObstacleCoursePath extends LinearOpMode {
...
-        // delete this
-        drive.setPoseEstimate(new Pose2d(5, 10, Math.toRadians(0)));
+        // first part
+        drive.setPoseEstimate(new Pose2d(-60, 60, Math.toRadians(270)));
+        Trajectory trajectoryGoForward = drive.trajectoryBuilder(new Pose2d(-60, 60, Math.toRadians(270)))
+                .splineTo(new Vector2d(-57.3, 29), Math.toRadians(270))
+                .build();
+        Trajectory trajectoryTurnLeft = drive.trajectoryBuilder(trajectoryFirstPart.end())
+                .splineTo(new Vector2d(5, 10), Math.toRadians(0))
+                .build();
```
Comment (k-mma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java:27]: Be consistent with spaces between each part--either add empty line before each comment for the trajectory parts or delete empty lines

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java [hunk 9]
```diff
@@ -52,6 +59,9 @@ public void runOpMode() {
 
         if(isStopRequested()) return;
 
+        // first chunk
```
Comment (k-mma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java:74]: If empty lines were added between the trajectory parts, consider adding spaces between the calling of the trajectory parts here.

**Review by k-mma — APPROVED** [review]

**Review by CubeOfCheese — APPROVED** [review]

**Pull request merged**